app/scripts/views/mixins/signin-mixin.js

```diff
@@ -55,19 +56,35 @@ define(function (require, exports, module) {
     },
 
     onSignInSuccess: function (account) {
-      if (! account.get('verified')) {
+      if (account.get('verified') && account.get('challenge') === Constants.SIGNIN_EMAIL) {
```

Comment:
Looking at the newest fxa-83 spec, [we'll have two `challenge` related fields](https://github.com/mozilla/fxa/pull/143/files#diff-a3580b834127f11689379869c2407b38R127): `challengeReason` and `challengeMethod`. We can use these two fields to make decisions on what to show the user. The two fields also give us flexibility to add additional factor types in the near future.

```js
if (! account.get('verified')) {
  var challengeType = account.get('challengeType');
  var challengeMethod = account.get('challengeMethod');
  if (challengeType === 'signup') { // any value in checking challengeMethod? 
                                   // Don't think so, but want to make sure
    return this.navigate('confirm'......
  } else if (challengeType === 'signin' && challengeMethod === 'email') {
    return this.navigate('confirm_token'...
  } else if (challengeType === 'signin' && challengeMethod === 'sms') {
     ....
  }
}
```

@philbooth - does that seem right to you? Now that we work through this, do two fields still seem worthwhile?